macosx/QuickLookPlugin/QuickLookPlugin-Prefix.pch

```diff
@@ -1,4 +1,5 @@
 #ifdef __OBJC__
 #import <Cocoa/Cocoa.h>
+#import <CoreFoundation/CFPlugInCOM.h>
```

Comment:
Is there any reason why importing the header only where required wouldn't suffice?

Reply:
This has been a tricky compilation issue on my Xcode.
When I started debugging the missing `QuickLookGeneratorPluginFactory` symbol issue, I attempted to convert main.cc to main.m: and I changed the main file `#include` to `#import`.

At the bottom of the file <QuickLook/QLGenerator.h>, there is this definition:
```
#if defined(__COREFOUNDATION_CFPLUGINCOM__)
typedef struct {
...
} QLGeneratorInterfaceStruct;
#endif
```
In <CoreFoundation/CFPlugInCOM.h>, there is:
```
#define __COREFOUNDATION_CFPLUGINCOM__ 1
```

Consequently:
- if <CoreFoundation/CFPlugInCOM.h> is imported before <QuickLook/QuickLook.h>, `QLGeneratorInterfaceStruct` exists.
- If <CoreFoundation/CFPlugInCOM.h> is imported after <QuickLook/QuickLook.h>, `QLGeneratorInterfaceStruct` doesn't exist.

But in Xcode, "QuickLookPlugin-Prefix.pch" is loaded first, and because those are `#import`, it wouldn't load the definition of `QLGeneratorInterfaceStruct` in main.m, and the file wouldn't compile anymore. To avoid such side effect misadventure again, I prefer to have `CFPlugInCOM.h` systematically imported before `QuickLook.h` in our QuickLookPlugin.

Regardless, in the end I kept `main.cc` as `main.cc`.

Reply:
Just a heads-up that TeamCity is down.

![SCR-20220428-kyn](https://user-images.githubusercontent.com/69029666/165668919-7366075c-13eb-4ba4-a65b-28bc4dae8766.png)